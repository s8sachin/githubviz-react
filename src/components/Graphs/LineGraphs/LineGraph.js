import React, { Component } from 'react';
import { XYPlot,XAxis,YAxis, VerticalGridLines, HorizontalGridLines,Hint,LineMarkSeries
} from 'react-vis';
import { Grid, Row,Col} from 'react-bootstrap';

const data1 = [
    {x: "Repo", y: "C1"},
    {x: "Repo1", y: "C2"},
    {x: "Repo2", y: "C3"},
    {x: "Repo3", y: "C4"},
    {x: "Repo4", y: "C5"},
    {x: "Repo5", y: "C6"},

  ];


class LineGraph extends Component {
    state = {
        value2: false
      }
  render() {
    const {value2} = this.state;
    return (
      <div>
      <Grid>
        <Row>
      <Col xs={12} md={12}>
      <XYPlot height={300} width={500} stroke="orange" xType="ordinal" yType="ordinal" >
        <VerticalGridLines />
        <HorizontalGridLines />
        <XAxis />
        <YAxis />
        <LineMarkSeries data={data1}
         onValueMouseOver={v => this.setState({ value2: v })}
         onSeriesMouseOut={v => this.setState({ value2: false })}
        />
         {value2 && <Hint value={value2} />}
      </XYPlot>
    </Col>
    </Row>
        </Grid>
        </div>
    );
  }
}

export default LineGraph;
