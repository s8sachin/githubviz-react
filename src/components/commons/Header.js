import React, { Component } from 'react';
import { Navbar, Nav, NavItem, MenuItem, NavDropdown, Button } from 'react-bootstrap';
import browserHistory from '../../history';

class Header extends Component {

  state = {
    userLoggedIn: false
  }
  
  componentWillMount () {
    this.setState({userLoggedIn: this.authenticate()})
  }

  authenticate () {
    return (localStorage.getItem('access_token') && localStorage.getItem('email')) ? true : false;
  }

  logout () {
    localStorage.removeItem('access_token');
    localStorage.removeItem('email');
    this.setState({userLoggedIn: false});
    browserHistory.push('/');
  }

  render () {
    return (
      <Navbar inverse collapseOnSelect>
        <Navbar.Header>
          <Navbar.Brand>
            <a href='#' id ="gitviz" onClick={() => browserHistory.push('/')}>Github Viz</a>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          {this.state.userLoggedIn && 
            <div>
              <Nav>
                <NavItem eventKey={1} href="#">
                  <Button  bsSize='xsmall' bsStyle="primary" onClick={() => browserHistory.push('/allGraphs')}>All Graphs</Button>
                </NavItem>
              </Nav>
              <Nav>
                <NavItem eventKey={2} href="#">
                  <Button  bsSize='xsmall' bsStyle="primary" onClick={() => browserHistory.push('/googleMap')}>Google Map</Button>
                </NavItem>
                {/* <NavItem eventKey={2} href="#">
                  Link
                  </NavItem>
                  <NavDropdown eventKey={3} title="Dropdown" id="basic-nav-dropdown">
                  <MenuItem eventKey={3.1}></MenuItem>
                  <MenuItem eventKey={3.2}>Another action</MenuItem>
                  <MenuItem eventKey={3.3}>Something else here</MenuItem>
                  <MenuItem divider />
                  <MenuItem eventKey={3.3}>Separated link</MenuItem>
                </NavDropdown> */}
              </Nav>
            </div>
          }
          <Nav pullRight>
            <NavItem eventKey={1} id="piechart" onClick={() => browserHistory.push('/piecharts')}>
             Pie chart
            </NavItem>
            <NavItem  id="linegrph" eventKey={2} onClick={() => browserHistory.push('/lineGraph')}>
              Line Gragh
            </NavItem>
            {this.state.userLoggedIn && <NavItem eventKey={2}>
              <Button bsSize='xsmall' bsStyle="primary" onClick={() => this.logout()}>Logout</Button>
            </NavItem>}
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    )
  }
}

export default Header;