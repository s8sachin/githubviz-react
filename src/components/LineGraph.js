import React, { Component } from 'react';
import Header from './commons/Header';
import 'react-vis/dist/style.css';
import { XYPlot,VerticalBarSeries,XAxis,YAxis, VerticalGridLines, HorizontalGridLines,Hint,LineMarkSeries
  } from 'react-vis';
  import { Grid, Row,Col} from 'react-bootstrap';
  const data = [
    {x: 0, y: 8},
    {x: 1, y: 5},
    {x: 2, y: 4},
    {x: 3, y: 9},
    {x: 4, y: 1},
    {x: 5, y: 7},
    {x: 6, y: 6},
    {x: 7, y: 3},
    {x: 8, y: 2},
    {x: 9, y: 0}
  ];
  const data1 = [
    {x: "Repo", y: "C1"},
    {x: "Repo1", y: "C2"},
    {x: "Repo2", y: "C3"},
    {x: "Repo3", y: "C4"},
    {x: "Repo4", y: "C5"},
    {x: "Repo5", y: "C6"},

  ];
class LineGraph extends Component {
  state = {
    value: false,
    value2: false
  }
  render () {
    const {value} = this.state;
    const {value2} = this.state;
    return (
      <div>
        <Header />
        <Grid>
          <Row className="show-grid">
            <Col xs={12} md={6}>
              <div>
                <XYPlot height={300} width={500} color="orange" stroke="black"  xType="ordinal">
                  <VerticalGridLines />
                  <HorizontalGridLines />
                  <XAxis />
                  <YAxis />
                  <VerticalBarSeries data={data}
                    onValueMouseOver={v => this.setState({ value: v })}
                    onSeriesMouseOut={v => this.setState({ value: false })} />
                  {value && <Hint value={value} />}
                </XYPlot>
                <h3 className="head-custom">BarGragh:</h3>
                <p className="para-custom">Bar series come in two flavors, HorizontalBarSeries and VerticalBarSeries</p>
              </div>
            </Col>
            <Col xs={12} md={6}>
              <XYPlot height={300} width={300} stroke="orange" xType="ordinal" yType="ordinal" >
                <VerticalGridLines />
                <HorizontalGridLines />
                <XAxis />
                <YAxis />
                <LineMarkSeries data={data1}
                 onValueMouseOver={v => this.setState({ value2: v })}
                 onSeriesMouseOut={v => this.setState({ value2: false })}
                />
                 {value2 && <Hint value={value2} />}
              </XYPlot>
              <h3 className="head-custom">LineGragh:</h3>
                <p className="para-custom">Line series come in two flavors, HorizontalBarSeries and VerticalBarSeries</p>
            </Col>
          </Row>
        </Grid>
      </div>
    )
  }
}
export default LineGraph;