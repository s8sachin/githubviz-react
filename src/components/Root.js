import React, { Component } from 'react';
import { Grid, Jumbotron, Button } from 'react-bootstrap';
import Header from './commons/Header';

class Root extends Component {

  clickFunc() {
    console.log("hi")
  }
  render() {
    return (
      <div>
        <Header/>
        <Grid bsClass='container'>
          <Jumbotron>
            <h1>Hello, world!</h1>
            <p>
              This is a simple hero unit, a simple jumbotron-style component for calling
              extra attention to featured content or information.
            </p>
            <p>
              <Button bsStyle="primary" href='#'>Learn more</Button>
            </p>
          </Jumbotron>
        </Grid>
      </div>
    );
  }
}

export default Root;