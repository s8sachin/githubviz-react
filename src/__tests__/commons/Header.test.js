import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Header from '../../components/commons/Header';
import Root from '../../components/Root';
import LocalStorageMock from '../../LocalStorageMock';
import { Link } from 'react-router';
import PieCharts from '../../components/piecharts';
import LineGraph from '../../components/LineGraph';

global.localStorage = new LocalStorageMock;


configure({ adapter: new Adapter() });

describe('Header', () => {
it('should mount Header component', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
   it("should click create github viz link", () => {
    const wrapper = mount(<Header />)
    // console.log(wrapper.instances());
    // expect(wrapper.instances().props.name).toBe('Header');
    // wrapper.find('abc').simulate('click');
    const tag = wrapper.find('#gitviz')
    // console.log(tag.simulate('click'))
    const output= tag.simulate('click')
    // const component = renderer.create(tag.props().onClick)
    const tree = shallow(<Root />)
    expect(toJson(tree)).toMatchSnapshot();
  })


  //using id
  it("should click create piechart link", () => {
    const wrapper = mount(<Header />)
    const tag1 = wrapper.find('#piechart').hostNodes();
    tag1.simulate('click')
    const tree1 = shallow(<PieCharts />)
    expect(toJson(tree1)).toMatchSnapshot();
  })


  it("should click create linegraph link", () => {
    const wrapper = mount(<Header />)
    const tag2 = wrapper.find('#linegrph').hostNodes();
    tag2.simulate('click')
    const tree2 =shallow(< LineGraph/>)
    expect(toJson(tree2)).toMatchSnapshot();
  })

  it("should click create login link", () => {
    const wrapper = mount(<Header />)
    const tag2 = wrapper.find('#linegrph').hostNodes();
    tag2.simulate('click')
    const tree2 =shallow(< LineGraph/>)
    expect(toJson(tree2)).toMatchSnapshot();
  })


  // it("should click create piechart link", () => {
  //   const wrapper = shallow(<Header />)
  //   const button = wrapper.find('Navbar').find('Navbar.Collapse').find('Nav').find('NavItem').first()
  //   const component = renderer.create(button.props().onClick)
  //   // let tree = component.toJSON();
  //   // tree.props.onClick();
  //   // expect(tree).toMatchSnapshot();
  // })



//   it("should click create a allgraph ", () => {
//     const wrapper = shallow(<Header />)
//    console.log(wrapper.instances());
//     // const Button = wrapper.find('Navbar').find('Nav').first().find('Button')
//     const Button = wrapper.find('my_id')
    
//     console.log(Button);
    // const component = renderer.create(card.props().link)
  //   let tree = component.toJSON();
  //   tree.props.onClick();
  //   expect(tree).toMatchSnapshot();
//   })

// it("should click create github viz link", () => {
//     const wrapper = render(<Header />)
//     // console.log(wrapper.instances());
//     // expect(wrapper.instances().props.name).toBe('Header');
//     // const tag = wrapper.find('Navbar').find('Navbar.Brand').first().find('a')
//     // const component = renderer.create(tag.props().href)
//     expect(wrapper.find(Link).props().to).toBe('/Header');
//     let tree = component.toJSON();
//     tree.props.onClick();
//     expect(tree).toMatchSnapshot();
//   })



// it("should click create github viz link", () => {
//     const wrapper = mount(<Header />)
//     // console.log(wrapper.instances());
//     // expect(wrapper.instances().props.name).toBe('Header');
//     // wrapper.find('abc').simulate('click');
//     const tag = wrapper.find('#abc')
//     tag.simulate('click')
//     // const component = renderer.create(tag.props().href)
//     // let tree = component.toJSON();
//     // tree.props.onClick();
//     expect(tag).toMatchSnapshot();
//   })
})