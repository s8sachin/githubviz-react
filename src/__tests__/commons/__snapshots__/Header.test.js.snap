// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should click create github viz link 1`] = `
<div>
  <Header />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Jumbotron
      bsClass="jumbotron"
      componentClass="div"
    >
      <h1>
        Hello, world!
      </h1>
      <p>
        This is a simple hero unit, a simple jumbotron-style component for calling extra attention to featured content or information.
      </p>
      <p>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          onClick={[Function]}
        >
          Login
        </Button>
      </p>
    </Jumbotron>
  </Grid>
</div>
`;

exports[`Header should click create linegraph link 1`] = `
<div>
  <Header />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      className="show-grid"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={6}
        xs={12}
      >
        <div>
          <XYPlot
            className=""
            color="orange"
            height={300}
            stroke="black"
            width={500}
            xType="ordinal"
          >
            <VerticalGridLines
              attr="x"
              direction="vertical"
            />
            <HorizontalGridLines
              attr="y"
              direction="horizontal"
            />
            <XAxis
              attr="x"
              attrAxis="y"
              orientation="bottom"
            />
            <YAxis
              attr="y"
              attrAxis="x"
              orientation="left"
            />
            <VerticalBarSeries
              className=""
              data={
                Array [
                  Object {
                    "x": 0,
                    "y": 8,
                  },
                  Object {
                    "x": 1,
                    "y": 5,
                  },
                  Object {
                    "x": 2,
                    "y": 4,
                  },
                  Object {
                    "x": 3,
                    "y": 9,
                  },
                  Object {
                    "x": 4,
                    "y": 1,
                  },
                  Object {
                    "x": 5,
                    "y": 7,
                  },
                  Object {
                    "x": 6,
                    "y": 6,
                  },
                  Object {
                    "x": 7,
                    "y": 3,
                  },
                  Object {
                    "x": 8,
                    "y": 2,
                  },
                  Object {
                    "x": 9,
                    "y": 0,
                  },
                ]
              }
              onSeriesMouseOut={[Function]}
              onValueMouseOver={[Function]}
              stack={false}
              style={Object {}}
            />
          </XYPlot>
          <h3
            className="head-custom"
          >
            BarGragh:
          </h3>
          <p
            className="para-custom"
          >
            Bar series come in two flavors, HorizontalBarSeries and VerticalBarSeries
          </p>
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={6}
        xs={12}
      >
        <XYPlot
          className=""
          height={300}
          stroke="orange"
          width={300}
          xType="ordinal"
          yType="ordinal"
        >
          <VerticalGridLines
            attr="x"
            direction="vertical"
          />
          <HorizontalGridLines
            attr="y"
            direction="horizontal"
          />
          <XAxis
            attr="x"
            attrAxis="y"
            orientation="bottom"
          />
          <YAxis
            attr="y"
            attrAxis="x"
            orientation="left"
          />
          <LineMarkSeries
            className=""
            curve={null}
            data={
              Array [
                Object {
                  "x": "Repo",
                  "y": "C1",
                },
                Object {
                  "x": "Repo1",
                  "y": "C2",
                },
                Object {
                  "x": "Repo2",
                  "y": "C3",
                },
                Object {
                  "x": "Repo3",
                  "y": "C4",
                },
                Object {
                  "x": "Repo4",
                  "y": "C5",
                },
                Object {
                  "x": "Repo5",
                  "y": "C6",
                },
              ]
            }
            getNull={[Function]}
            lineStyle={Object {}}
            markStyle={Object {}}
            onSeriesMouseOut={[Function]}
            onValueMouseOver={[Function]}
            opacity={1}
            stack={false}
            strokeStyle="solid"
            style={Object {}}
          />
        </XYPlot>
        <h3
          className="head-custom"
        >
          LineGragh:
        </h3>
        <p
          className="para-custom"
        >
          Line series come in two flavors, HorizontalBarSeries and VerticalBarSeries
        </p>
      </Col>
    </Row>
  </Grid>
</div>
`;

exports[`Header should click create login link 1`] = `
<div>
  <Header />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      className="show-grid"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={6}
        xs={12}
      >
        <div>
          <XYPlot
            className=""
            color="orange"
            height={300}
            stroke="black"
            width={500}
            xType="ordinal"
          >
            <VerticalGridLines
              attr="x"
              direction="vertical"
            />
            <HorizontalGridLines
              attr="y"
              direction="horizontal"
            />
            <XAxis
              attr="x"
              attrAxis="y"
              orientation="bottom"
            />
            <YAxis
              attr="y"
              attrAxis="x"
              orientation="left"
            />
            <VerticalBarSeries
              className=""
              data={
                Array [
                  Object {
                    "x": 0,
                    "y": 8,
                  },
                  Object {
                    "x": 1,
                    "y": 5,
                  },
                  Object {
                    "x": 2,
                    "y": 4,
                  },
                  Object {
                    "x": 3,
                    "y": 9,
                  },
                  Object {
                    "x": 4,
                    "y": 1,
                  },
                  Object {
                    "x": 5,
                    "y": 7,
                  },
                  Object {
                    "x": 6,
                    "y": 6,
                  },
                  Object {
                    "x": 7,
                    "y": 3,
                  },
                  Object {
                    "x": 8,
                    "y": 2,
                  },
                  Object {
                    "x": 9,
                    "y": 0,
                  },
                ]
              }
              onSeriesMouseOut={[Function]}
              onValueMouseOver={[Function]}
              stack={false}
              style={Object {}}
            />
          </XYPlot>
          <h3
            className="head-custom"
          >
            BarGragh:
          </h3>
          <p
            className="para-custom"
          >
            Bar series come in two flavors, HorizontalBarSeries and VerticalBarSeries
          </p>
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={6}
        xs={12}
      >
        <XYPlot
          className=""
          height={300}
          stroke="orange"
          width={300}
          xType="ordinal"
          yType="ordinal"
        >
          <VerticalGridLines
            attr="x"
            direction="vertical"
          />
          <HorizontalGridLines
            attr="y"
            direction="horizontal"
          />
          <XAxis
            attr="x"
            attrAxis="y"
            orientation="bottom"
          />
          <YAxis
            attr="y"
            attrAxis="x"
            orientation="left"
          />
          <LineMarkSeries
            className=""
            curve={null}
            data={
              Array [
                Object {
                  "x": "Repo",
                  "y": "C1",
                },
                Object {
                  "x": "Repo1",
                  "y": "C2",
                },
                Object {
                  "x": "Repo2",
                  "y": "C3",
                },
                Object {
                  "x": "Repo3",
                  "y": "C4",
                },
                Object {
                  "x": "Repo4",
                  "y": "C5",
                },
                Object {
                  "x": "Repo5",
                  "y": "C6",
                },
              ]
            }
            getNull={[Function]}
            lineStyle={Object {}}
            markStyle={Object {}}
            onSeriesMouseOut={[Function]}
            onValueMouseOver={[Function]}
            opacity={1}
            stack={false}
            strokeStyle="solid"
            style={Object {}}
          />
        </XYPlot>
        <h3
          className="head-custom"
        >
          LineGragh:
        </h3>
        <p
          className="para-custom"
        >
          Line series come in two flavors, HorizontalBarSeries and VerticalBarSeries
        </p>
      </Col>
    </Row>
  </Grid>
</div>
`;

exports[`Header should click create piechart link 1`] = `
<div>
  <Header />
  <Radialcharts />
  <ClockExample />
</div>
`;

exports[`Header should mount Header component 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  collapseOnSelect={true}
  inverse={true}
>
  <NavbarHeader>
    <NavbarBrand>
      <a
        href="#"
        id="gitviz"
        onClick={[Function]}
      >
        Github Viz
      </a>
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}
    >
      <NavItem
        active={false}
        disabled={false}
        eventKey={1}
        id="piechart"
        onClick={[Function]}
      >
        Pie chart
      </NavItem>
      <NavItem
        active={false}
        disabled={false}
        eventKey={2}
        id="linegrph"
        onClick={[Function]}
      >
        Line Gragh
      </NavItem>
    </Nav>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;
